### ISSUE TITLE
How can I check if http header is set properly in the request?

### ISSUE THREAD

**Issue by postacik** [alexmercerind/media_kit#306]
I recently upgraded media_kit to the latest version (1.1.0) in my project.

I set an authentication cookie while constructing the Media object as in the following code:

```dart
    _player.open(Playlist([
      Media(url, httpHeaders: {"Cookie": KabloTVService.kablowebtvCookie})
    ]));
```

This works fine on MacOS but media (url) does not load in Windows.

It worked fine both on Windows and MacOS with the 0.0.7 version  of media_kit.

How can I check if the http header is set properly while opening the url?

Is there a difference between the MacOS and Windows library implementations that may cause this problem?

**Comment by postacik** [comment]
Is there any network inspector you can suggest to use on Windows?

**Comment by alexmercerind** [comment]
@postacik,

Our unit-tests for HTTP headers pass on Windows:

https://github.com/alexmercerind/media_kit/actions/runs/5653256024/job/15314144157

- `player-open-playable-media-http-headers`
- `player-open-playable-playlist-http-headers`

<hr>

I'll wait.

https://github.com/alexmercerind/media_kit/blob/f2918c0a2223aa4fe8434127b3969c016ba40eac/media_kit/test/src/player/player_test.dart#L565-L738

**Comment by alexmercerind** [comment]
You can try to pin your media_kit_libs_windows_video version till then. Also, does this relate to #307 in any way?

**Comment by alexmercerind** [comment]
It's not related to HTTP headers at all AFAIK. Share sample URL.

**Comment by Vocatis** [comment]
> You can try to pin your media_kit_libs_windows_video version till then. Also, does this relate to #307 in any way?

I do use headers without any issues, on both _media_kit_libs_windows_video_ 1.0.2 & 1.0.4, not the '**Cookie**' one, though.

**Comment by postacik** [comment]
@alexmercerind , 

I pinned media_kit_libs_windows_video to 1.0.2 with the following line: (I suppose that's what you mean)

```yaml
media_kit_libs_windows_video: 1.0.2
```

then after...

```
flutter clean
flutter pub get 
flutter run
```

...the url opened without any issues...

**Comment by alexmercerind** [comment]
Obviously, downgrading will work, but that does not provide the solution.

Also, your report is wrong. HTTP headers were always working in our unit-tests.

**Comment by alexmercerind** [comment]
Subscribe to #307.

**Comment by postacik** [comment]
I agree, HTTP headers is not the suspect in my case. 

It was just my first assumption because in my app, the video files need an authentication cookie to play and since they played fine on MacOS but failed on Windows, I thought the cookie was not being sent in the header for Windows. I'll follow #307 as you suggest...
